MoreLinq/CountMethods.cs

```diff
@@ -205,22 +191,22 @@ static bool QuantityIterator<T>(IEnumerable<T> source, int limit, int min, int m
 
                 return Convert.ToInt32(firstHasNext).CompareTo(Convert.ToInt32(secondHasNext));
             }
+        }
 
-            int PartialCount<T>(IEnumerable<T> source, int limit)
-            {
-                var count = 0;
+        static int PartialCount<T>(IEnumerable<T> source, int limit)
```

Comment:
- Convert `PartialCount` to an extension method so it's reusable but leave it private as it is now
- Rename `PartialCount` to `CountUpto`
- Like for all extension methods, check `source` is non-null
- Rename `limit` to `max`
- Change implementation so it behaves well for `max` ≤ 0 and perhaps even validate that `max` is never negative.